### PR TITLE
Use original logrus

### PR DESCRIPTION
**Pull request by ltachet**
theone2

# Use sirupsen/logrus instead of Vend fork

### Why?

Originally, the logrus fork has been created because of this issue: https://github.com/Sirupsen/logrus/issues/377.
But a fix has been merged since some time.

After having a lot of dependencies issues in another library (see https://github.com/vend/db/pull/32), we thought about removing vend/logrus little by little. I'm starting with this repo because it's often because of it that we have this a dependency to `vend/logrus`.

### Concerning this PR

- While updating the dependency, I have also updated the go version to the last one.
- The tests are not working without the flag `-gcflags=-l` which disable inlining. This is because we are checking the order of the stack trace (see difference between first and second commit of this PR). 

I detected that because the tests where _always_ passing with the debugger (`dlv test -- -test.v`) and _never_ with `go test`.

Here are the stack trace packages in the function `calcSkipStackFrames` which removes stack frames related to log: 

```sh
# WITH -gcflags=-l
github.com/vend/logrus-bugsnag (*bugsnagHook).Fire
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus LevelHooks.Fire
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus (*Entry).fireHooks
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus Entry.log
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus (*Entry).Log
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus (*Entry).Error
github.com/vend/logrus-bugsnag TestNoticeReceived
testing tRunner # first package without 'logrus-bugsnag', previous one is the top

# WITHOUT -gcflags=-l
github.com/vend/logrus-bugsnag (*bugsnagHook).Fire
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus LevelHooks.Fire
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus (*Entry).fireHooks
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus Entry.log
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus (*Entry).Log
github.com/vend/logrus-bugsnag TestNoticeReceived
github.com/vend/logrus-bugsnag/vendor/github.com/sirupsen/logrus (*Entry).Error
testing tRunner # first package without 'logrus-bugsnag', previous one is the top
```

So we could either:

1. remove the tests about the stack frame which are not very relevant if they depend on the compiler optimisations,

2. consider that the test is important because it allows to test the `calcSkipStackFrames` function.

But because bugsnag is going to be removed progressively, I don't think that future tests are going to be added here. So I just choose the easiest method and added the flags in travis...

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,11 @@
 language: go
+go: "1.12.x"
 
-go:
-  - 1.4
```
Comment (edsrzf) [inline .travis.yml:4]: 👴

File: ci/linter.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+
+# Install and run golangci-lint v1.17.1.
+# Last golangci-lint version is available here: https://github.com/golangci/golangci-lint/releases
+# Linter will display error for information, but will not make the build fail.
+
+curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(go env GOPATH)/bin v1.17.1
+golangci-lint run || true
```
Comment (edsrzf) [inline ci/linter.sh]: I assume we're getting rid of this soon, as before 😉

**Review by edsrzf — APPROVED** [review]
Changes make sense. This will make the package no longer compatible with `vend/logrus`, right? So this means we need to update `vend/log` pretty soon to avoid much breakage?

File: .travis.yml [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+# Skip the installation step
+install: true
+
+# Inlining in disable in tests in order to test the stack trace correctly.
+script:
+  - ./ci/build.sh || travis_terminate 1
+  - go test -v ./... -gcflags=-l
```
Comment (edsrzf) [inline .travis.yml:10]: Wow, tricky

**Comment by snikch** [comment]
FYI Cardinal is using this branch of `vend/log` which uses a branch of `vend/logrus` that had a current (at the time) merge of master from `sirupson/logrus`. It's been doing it for quite some time now, which makes me confident in our ability to get on mainline logrus 🤙 

https://github.com/vend/log/pull/31

File: ci/linter.sh (hunk 2 above)
Comment (ltachet) [reply ci/linter.sh]: Oops, I copy / pasted the file 😬
I'll remove that before merging (and fix any linting error if necessary)

File: .travis.yml (hunk 3 above)
Comment (ltachet) [reply .travis.yml:10]: Yes, tooks me a little time to understand why it was working while I was trying to debug and not when running the tests 😅

**Review by snikch — APPROVED** [review]

**Pull request merged**